File: solutions/2899. Last Visited Integers/2899.java
class Solution {
  public List<Integer> lastVisitedIntegers(List<String> words) {
    List<Integer> ans = new ArrayList<>();
    List<Integer> nums = new ArrayList<>();
    int k = 0;

    for (final String word : words)
      if (word.equals("prev")) {
        ++k;
        ans.add(k > nums.size() ? -1 : nums.get(nums.size() - k));
      } else {
        k = 0;
        nums.add(Integer.valueOf(word));
      }

    return ans;
  }
}

File: solutions/2899. Last Visited Integers/2899.py
class Solution:
  def lastVisitedIntegers(self, words: list[str]) -> list[int]:
    ans = []
    nums = []
    k = 0

    for word in words:
      if word == 'prev':
        k += 1
        ans.append(-1 if k > len(nums) else nums[-k])
      else:
        k = 0
        nums.append(int(word))

    return ans
